src/Features/CSharp/Portable/CodeRefactorings/AddMissingImports/CSharpAddMissingImportsRefactoringProvider.cs

```diff
@@ -19,7 +17,7 @@ internal class CSharpAddMissingImportsRefactoringProvider : AbstractAddMissingIm
 
         [ImportingConstructor]
         [SuppressMessage("RoslynDiagnosticsReliability", "RS0033:Importing constructor should be [Obsolete]", Justification = "Used in test code: https://github.com/dotnet/roslyn/issues/42814")]
-        public CSharpAddMissingImportsRefactoringProvider(IPasteTrackingService pasteTrackingService)
+        public CSharpAddMissingImportsRefactoringProvider([Import(AllowDefault = true)] IPasteTrackingService? pasteTrackingService)
```

Comment:
📝 Refactorings are exported with MEF attributes, but they are directly constructed by the infrastructure and not allowed to have any arguments to the constructor.

Reply:
@sharwell But this feature works, doesn't it?

Reply:
Yes, because it's provided through the MEF catalog in VS. If we move refactorings to OOP, it will stop working.

Reply:
not sure what "directly constructed by the infrastructure" means here.  The only place we use this is either in a mef host, or directly calling from unit tests.  Both work fine for the case of having constructor args.  And, afaict, all this is doing is making the arg optional if hte mef container doens't have it.  It's otherwise identical to before.

Reply:
This featuer is currently inherently tied to VS (it depends on mutable state tracked in ITextBuffers :)).  It's not moving OOP without major rework on how it operates/communicates :)